vertx-web-client/src/main/java/io/vertx/ext/web/client/predicate/ErrorConverter.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.vertx.ext.web.client.predicate;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.ext.web.client.impl.predicate.ErrorConverterImpl;
+
+import java.util.function.Function;
+
+/**
+ * Converts a {@link ResponsePredicateResult} to a {@code Throwable} describing the error.
+ *
+ * @author Thomas Segismont
+ */
+@VertxGen
+public interface ErrorConverter {
```

Comment:
this can extend `Function<ResponsePredicateResult, Throwable> converter` to be also a functional interface and declare a `bufferBody()` method that returns `false` by default, so one can create a converter simpler using a lambda.

Reply:
Done